causalml/inference/meta/xlearner.py

```diff
@@ -570,21 +655,30 @@ def fit(self, X, treatment, y):
             self.models_tau_c[group].fit(X_filt[w == 0], d_c)
             self.models_tau_t[group].fit(X_filt[w == 1], d_t)
 
-    def predict(self, X, p, treatment=None, y=None, return_components=False, verbose=True):
+    def predict(self, X, treatment=None, y=None, p=None, return_components=False,
+                return_p_score=False, verbose=True):
         """Predict treatment effects.
 
         Args:
             X (np.matrix or np.array or pd.Dataframe): a feature matrix
-            p (np.ndarray or pd.Series or dict): an array of propensity scores of float (0,1) in the single-treatment
-                case; or, a dictionary of treatment groups that map to propensity vectors of float (0,1)
             treatment (np.array or pd.Series, optional): a treatment vector
             y (np.array or pd.Series, optional): an outcome vector
-
+            p (np.ndarray or pd.Series or dict, optional): an array of propensity scores of float (0,1) in the
+                single-treatment case; or, a dictionary of treatment groups that map to propensity vectors of
+                float (0,1); if None will run ElasticNetPropensityModel() to generate the propensity scores.
+            return_components (bool, optional): whether to return outcome for treatment and control seperately
+            return_p_score (bool, optional): whether to return propensity score
+            verbose (bool, optional): whether to output progress logs
         Returns:
             (numpy.ndarray): Predictions of treatment effects.
         """
         X, treatment, y = convert_pd_to_np(X, treatment, y)
-        check_p_conditions(p, self.t_groups)
+
+        if p is None:
+            p = self.propensity
```

Comment:
As discussed, we need to generate `p` using `self.propensity_model` trained during `fit()` instead of using `self.propensity` because `X` in `predict()` might not be the same in `X` in `fit()`.